### ISSUE TITLE
Adaption fails with segmentation error

### ISSUE THREAD

**Issue by azacharos** [sparta/sparta#215]
## Summary

I tried to adapt a two level grid and the code returns segmentation error.

## Type of Issue

Bug Report

## SPARTA Version (Bug Report)

It appears on both latest stable versions (20Nov2020). 

## Expected Behavior (Bug Report)

It is expected to refine the grid. 

## Actual Behavior (Bug Report)

I tried the "static" option with the adapt command and the "on the fly" option with the fix/adapt command. The options that i used can be found below.  I tried also to reduce the refinement region by using a grid group, but the result was the same. 
adapt_grid    all refine particle 10 4 
fix     20 adapt 20 all refine coarsen particle 10 4 

## Steps to Reproduce (Bug Report)
My grid size is 10million cells. 

## Further Information, Files, and Links
[free_jet.zip](https://github.com/sparta/sparta/files/5765795/free_jet.zip)

**Comment by stanmoore1** [comment]
Hi @azacharos thanks for the bug report. Can you please attach the files needed to reproduce this issue? (The smaller the better).

**Comment by azacharos** [comment]
Hi, 
I attached the zip file that contains a smaller case that appears the same problem after 5000 time steps. 
I am using the mpi version of the code with the flag DSPARTA_BIGBIG on 12 cpu cores.

**Comment by stanmoore1** [comment]
Thank you, I'll take a look.

**Comment by stanmoore1** [comment]
I can reproduce a segfault if I use this in your reproducer:

```
run                 2000
adapt_grid all refine particle 10 4

write_restart       flow.restart.final
```

**Comment by azacharos** [comment]
The error appears also if the value option is used in the adapt_grid.

**Comment by stanmoore1** [comment]
Yeah, I am debugging now, should have a fix soon.

**Comment by stanmoore1** [comment]
The issue is that `childlist` here https://github.com/sparta/sparta/blob/master/src/adapt_grid.cpp#L503 is too small. The `nx` `ny` and `nz` default to 2x2x1 unless you use the `cells` keyword. I may need help from @sjplimp to get the logic right.

**Comment by stanmoore1** [comment]
As a workaround, you can set `childlist` bigger. You need a factor of 100 (since 20x20x1 divided by 2x2x1 = 100).

```patch
$ git diff adapt_grid.cpp
diff --git a/src/adapt_grid.cpp b/src/adapt_grid.cpp
index 961c993..b72a43c 100644
--- a/src/adapt_grid.cpp
+++ b/src/adapt_grid.cpp
@@ -500,7 +500,7 @@ void AdaptGrid::setup(int iter)
 
   // list of new cell indices for one refined cell
 
-  childlist = new int[nx*ny*nz];
+  childlist = new int[nx*ny*nz*100];
 
   // rlist and clist for refine/coarsen
 ```

**Comment by stanmoore1** [comment]
[adapt_bug.zip](https://github.com/sparta/sparta/files/5772604/adapt_bug.zip)

Here is a more minimal reproducer.

**Comment by stanmoore1** [comment]
Fixed by #224, closing #215.
